Kubernetes/helmcharts/azure-policy-addon-arc-connected-clusters/templates/remove-addon.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: pre-delete-admin
+  namespace: kube-system
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "20"
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: pre-delete-agent
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "20"
+rules:
+- apiGroups: ["apps"]
+  resources: ["deployments"]
+  verbs: ["delete"]
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: pre-delete-agent
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "20"
+subjects:
+- kind: ServiceAccount
+  name: pre-delete-admin
+  namespace: kube-system
+roleRef:
+  kind: Role
+  name: pre-delete-agent
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: addon-delete
+  namespace: kube-system
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": "hook-succeeded"
+spec:
+  template:
+    spec:
+      serviceAccountName: pre-delete-admin
+      automountServiceAccountToken: true 
+      restartPolicy: Never
+      terminationGracePeriodSeconds: 0
+      containers:
+        - name: pre-delete
+          image: "appropriate/curl" # alpine + curl
+          imagePullPolicy: IfNotPresent
+          command: 
+            - "/bin/sh"
+            - "-ec"
+            - |
+              set -o pipefail
+              TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+              curl -s -X DELETE -k https://kubernetes.default.svc/apis/apps/v1/namespaces/kube-system/deployments/azure-policy \
```

Comment:
the namespace should probably be a variable

Reply:
@EmandM Can you confirm with AKS team if calling Kubernetes service using https://kubernetes.default.svc is a supported way.

Reply:
https://kubernetes.default.svc is the default way to access the kubernetes api server from inside any container running on a kubernetes cluster. That doesn't change even if the api server is managed. The permissions for this api are managed via service accounts, which is why we install a service account along with the hook.

Reply:
Updated namespace to be a variable